plugins/org.eclipse.emf.ecoretools.ale.core/src/org/eclipse/emf/ecoretools/ale/core/interpreter/IAleInterpreter.java

```diff
@@ -0,0 +1,110 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Inria - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.emf.ecoretools.ale.core.interpreter;
+
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecoretools.ale.core.env.ClosedAleEnvironmentException;
+import org.eclipse.emf.ecoretools.ale.core.env.IAleEnvironment;
+import org.eclipse.emf.ecoretools.ale.core.interpreter.internal.AleEngine;
+import org.eclipse.emf.ecoretools.ale.implementation.Method;
+import org.eclipse.sirius.common.tools.api.interpreter.IInterpreterWithDiagnostic.IEvaluationResult;
+
+/**
+ * Instances of this class are able to execute an {@link EObject} while taking
+ * ALE behaviors into account.
+ * <p> 
+ * Use {@link IAleEnvironment#getInterpreter()} to get a reliable interpreter.
+ * <p>
+ * Interpreters must be closed to prevent memory leaks. Interpreters delivered by
+ * an environment {@link IAleEnvironment#close() do not have to be closed manually}.
+ */
+public interface IAleInterpreter extends AutoCloseable {
+    
+	/**
+	 * Returns the logger used by the interpreter.
+	 * <p>
+	 * Can be used to access stacktraces.
+	 * 
+	 * @return the logger used by the interpreter.
+	 */
+    DiagnosticLogger getLogger();
+	
+	/**
+	 * Calls the {@code main} method onto {@code caller}, passing {@code args}.
+	 * <p>
+	 * <strong>Caution</strong>: all objects must belong to the same environment.
+	 * {@code caller} must have been 
+	 * {@link IAleEnvironment#loadModel(org.eclipse.emf.common.util.URI) loaded from the environment}
+	 * and {@code main} must come from 
+	 * {@link IAleEnvironment#getBehaviors() environment's behaviors}.
+	 * 
+	 * @param caller
+	 * 			The object on which the main method will be called.
+	 * @param main
+	 * 			The method to call on the caller.
+	 * @param args
+	 * 			The arguments that will be passed to the method.
+	 * @return
+	 */
+	IEvaluationResult eval(EObject caller, Method main, List<Object> args) throws ClosedAleEnvironmentException;
+
+	/**
+	 * Registers services for each methods from a list of qualified class names.
+	 * 
+	 * @param services
+	 * 			The new services to register.
+	 */
+	void registerServices(Iterable<String> services);
+
+	/**
+	 * Declare plug-ins & projects where services can be found.
+	 * 
+	 * @param plugins
+	 * 			The names of the plug-ins providing services.
+	 * @param project
+	 * 			The names of the plug-ins providing services.
+	 */
+	void initScope(Set<String> plugins, Set<String> project);
+    
+    /**
+     * Returns the engine used by the ongoing evaluation.
+     * @return the engine if the evaluation has started, null otherwise
+     * 
+     * @deprecated does not seem to be used
+     */
+	@Deprecated
+	AleEngine getCurrentEngine();
```

Comment:
This method is used by GEMOC ALEEngine  (see https://github.com/eclipse/gemoc-studio-execution-ale/blob/67d478837fa746294d0bb2bff7921282729ad733/plugins/org.eclipse.gemoc.ale.interpreted.engine/src/org/eclipse/gemoc/ale/interpreted/engine/AleEngine.java#L138)

not sure yet if it can be replaced/simplified by the API improvement you proposed

Reply:
after verification on GEMOC side, it seems that the new API on AleInterpreter should do the job

Reply:
I suggest to keep it for now and remove it only once https://github.com/eclipse/gemoc-studio-execution-ale/ is updated with latest changes.